woocommerce/woocommerce-android
# Count Orders instead of retrieving all Orders to display the badge

Do not merge until [this PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2429/files) in FluxC gets merged 

Closes: #6658

### Description
This PR updates how we obtain the `Processing Orders` from the DB to display the Order's badge in the bottom navigation control. It does that by using the function `observeOrderCountForSite` and relying on SQLite `COUNT` instead of retrieving the list of orders to check the size.

### Testing instructions

1. Open the app and check that the number of `Processing Orders` matches the value of the badge in the bottom navigation control.
2. Select an order with a status different than `Processing`.
3. Update the order status to `Processing`.
4. Check that the value of the badge in the bottom navigation control increased by one matching the number of `Processing Orders`
5. Select the same order we used before
6. Update the status to other than `Processing`.
7. Check that the value in the badge in the bottom navigation control decreased by one, matching the number of `Processing Orders`

### Images/gif

https://user-images.githubusercontent.com/18119390/170832987-fa673e73-9859-4641-8ea9-ab4dae794d31.mp4
